RetailCoder.VBE/Settings/CodeInspectionConfigProvider.cs

```diff
@@ -1,41 +1,43 @@
 ﻿using System.Collections.Generic;
-using Rubberduck.Parsing.Inspections.Resources;
+using System.Linq;
 using Rubberduck.SettingsProvider;
 using Rubberduck.Parsing.Inspections.Abstract;
-using System.Linq;
 
 namespace Rubberduck.Settings
 {
     public class CodeInspectionConfigProvider : IConfigProvider<CodeInspectionSettings>
     {
         private readonly IPersistanceService<CodeInspectionSettings> _persister;
-        private readonly IEnumerable<IInspection> _foundInspections;
+        private readonly CodeInspectionSettings _defaultSettings;
 
         public CodeInspectionConfigProvider(IPersistanceService<CodeInspectionSettings> persister, IEnumerable<IInspection> foundInspections)
         {
             _persister = persister;
-            _foundInspections = foundInspections;
+            _defaultSettings = new DefaultSettings<CodeInspectionSettings>().Default;
+
+            var nonDefaultInspections = foundInspections
+                .Where(inspection => !_defaultSettings.CodeInspections.Select(x => x.Name).Contains(inspection.Name));
```

Comment:
I think that's iterating the inspections twice. `.Any()` could help here I think:

```csharp
var nonDefaultInspections = foundInspections
    .Where(inspection => !_defaultSettings.CodeInspections.Any(x => x.Name == inspection.Name));
```

Reply:
`Any` also iterates, so I extracted the names to a HashSet to avoid this.

Reply:
It does, but `Select` is first projecting all elements into an `IEnumerable<string>`, which `Contains` then iterates. `Any` skips the projection part; HashSet seems overkill.

Reply:
This part got removed.